### PR TITLE
switch BoxOperations.selectTop to use wallets BoxSelectors.select;

### PR DESCRIPTION
**Pull request by greenhat**
ergo PR - https://github.com/ergoplatform/ergo/pull/1071

Todo:
- [x] use new box selection in ergodex;
- [x] use tx building from wallet;
- [x] use tx building in ergodex https://github.com/ergoplatform/ergo-dex/pull/14; 
- [x] use box selection and tx building in ergotool https://github.com/ergoplatform/ergo-tool/pull/53;

### CODE REVIEW

File: common/src/main/java/org/ergoplatform/appkit/JavaHelpers.scala [hunk 1]
```diff
@@ -77,6 +78,22 @@ object Iso extends LowPriorityIsos {
...
+        .map(t => bytesToId(t._1) -> t._2)
+        .toMap
+    }
+    override def from(t: Map[ModifierId, Long]): JList[ErgoToken] = {
+      import JavaHelpers._
+      val pairs: IndexedSeq[(TokenId, Long)] = t.toIndexedSeq
+        .map(t => (Digest32 @@ idToBytes(t._1)) -> t._2)
+      pairs.convertTo[JList[ErgoToken]]
+    }
+  }
```
Comment (aslesarenko) [inline common/src/main/java/org/ergoplatform/appkit/JavaHelpers.scala:95]: Please use LinkedListMap

Comment (greenhat) [reply common/src/main/java/org/ergoplatform/appkit/JavaHelpers.scala:96]: Oh, I missed this one. Thanks! Fixed.

Comment (greenhat) [reply common/src/main/java/org/ergoplatform/appkit/JavaHelpers.scala:96]: BTW, I assumed you meant `LinkedHashMap`, cause I did not find `LinkedListMap`.

Comment (aslesarenko) [reply common/src/main/java/org/ergoplatform/appkit/JavaHelpers.scala:96]: yes, sure

**Pull request merged**